app/registries/foreman/plugin.rb

```diff
@@ -372,6 +372,11 @@ def register_facet(klass, name, &block)
       @facets << Facets.register(klass, name, &block)
     end
 
+    def register_info_provider(klass)
+      @info_providers ||= []
+      @info_providers << HostInfo.register_info_provider(klass)
```

Comment:
Current registry scheme enables registering providers from plugins and from arbitrary places in the code (initializers for example).
The concept is that `HostInfo` caches providers from plugins on first access and then you can add additional providers. If the cache is already committed, we need to add the providers directly, and save them for later cache refresh.